src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/TableConcatenate.txt

```diff
@@ -0,0 +1,29 @@
+
+>> TableConcatenate()
+Table()
+
+>> TableConcatenate([{a:0, b:false, c:"Hello"}], [{a:1, b:true, c:"World"}])
+Table({a:0,b:false,c:"Hello"},{a:1,b:true,c:"World"})
+
+>> TableConcatenate([{a:0}], [{a:true}], [{a:2}])
+Table({a:0},{a:1},{a:2})
+
+>> TableConcatenate([{a:0, b:"hello"}], [{a:true}], [{a:2}])
+Table({a:0,b:"hello"},{a:1,b:Blank()},{a:2,b:Blank()})
+
+>> TableConcatenate([{a:0}], [{b:true}], [{c:"Hello"}])
+Table({a:0,b:Blank(),c:Blank()},{a:Blank(),b:true,c:Blank()},{a:Blank(),b:Blank(),c:"Hello"})
+
+>> TableConcatenate([{a:0}], [{b:true}], [{c:"Hello", d: {x: "World"}}])
+Table({a:0,b:Blank(),c:Blank(),d:Blank()},{a:Blank(),b:true,c:Blank(),d:Blank()},{a:Blank(),b:Blank(),c:"Hello",d:{x:"World"}})
+
+// Invalid input - record instead of table 
+>> TableConcatenate({c:"Hello", d: {x: "World"}}) 
+Errors: Error 17-45: Invalid argument type (Record). Expecting a Table value instead.|Error 0-46: The function 'TableConcatenate' has some invalid arguments.
+
+// Invalid input - trying to coerce guid and datetime 
+>> TableConcatenate([{a:Date(2024,1,1)}], [{a:GUID("some-guid-value-1234")}]) 
+Errors: Error 39-73: Incompatible type. The item you are trying to put into a table has a type that is not compatible with the table.|Error 0-74: The function 'TableConcatenate' has some invalid arguments.
+
+>> TableConcatenate(Blank())
+Table(Blank())
```

Comment:
>Table [](http://example.com/codeflow?start=0&length=5)

Additional interesting tests:

- TC( [1, 2], If(1<0, Table({Value:{a:2}})) )
    - The second argument is a _typed blank table_, but the schema is incompatible. It should fail (at compile-time) even if the second argument is blank.
- TC( [1, 2], Blank(), [4, 5], Blank(), [7, 8] )
    - Multiple blank arguments
- TC( [1,2], If(1/0<2,[3,4]), [5,6] )
    - Tests with error arguments
    - I think we can return an error if any of the tables are erroneous. That is consistent with other functions that take multiple tables
- TC( [1, 2, 3/0, 4], [5, Sqrt(-1), 7, 8] )
    - Tests with errors inside property values
    - Those are valid records, it should work
    - Result will be Table({Value:1},{Value:2},{Value:Error({Kind:ErrorKind.Div0}),{Value:4},...}
- TC( Filter([2,1,0,-1,-2], 1/Value>0), Filter([-2,-1,0,1,2], Log(Value)>0) )
    - Tests with error records
    - Those are valid tables